pypsa/optimization/abstract.py

```diff
@@ -248,3 +248,30 @@ def optimize_security_constrained(
                 m.add_constraints(lhs, sign, rhs, name=constraint + "-security")
 
     return n.optimize.solve_model(**kwargs)
+
+
+def optimize_with_rolling_horizon(n, snapshots=None, horizon=100, **kwargs):
+    """
+    Optimizes the network in a rolling horizon fashion.
+
+    Parameters
+    ----------
+    n : pypsa.Network
+    snapshots : list-like
+        Set of snapshots to consider in the optimization. The default is None.
+    horizon : int
+        Number of snapshots to consider in each iteration. Defaults to 100.
+    **kwargs:
+        Keyword argument used by `linopy.Model.solve`, such as `solver_name`,
+
+    Returns
+    -------
+    None
+    """
+    if snapshots is None:
+        snapshots = n.snapshots
+
+    n_snapshots = len(snapshots)
+    for i in range(0, len(snapshots), horizon):
+        n.optimize(snapshots[i : min(n_snapshots, i + horizon)], **kwargs)
```

Comment:
One should pass the state of charges from previous optimisation horizons:

https://github.com/PyPSA/pypsa-eur/pull/204/files#diff-5c213776d6b3578d6249226cef589c81d6f7846a0c7e2f22d05da726400d3875

Reply:
thanks, will take care of it